autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -806,37 +798,122 @@ def get_inputs_to_stacker(self, X, level_start, level_end, model_levels=None, y_
+                model = self.load_model(model)
+                model_name = model.name
+                model_compressed = model.convert_to_refitfull_template()
+                # TODO: Do it for all models in the level at once to avoid repeated processing of data?
+                stacker_type = type(model)
+                if issubclass(stacker_type, WeightedEnsembleModel):
+                    # TODO: Technically we don't need to re-train the weighted ensemble, we could just copy the original and re-use the weights.
+                    if self.bagged_mode:
+                        X_train_stack_preds = self.get_inputs_to_stacker(X, level_start=0, level_end=level, fit=True)
+                        y_input = y
+                    else:
+                        X_train_stack_preds = self.get_inputs_to_stacker(X_val, level_start=0, level_end=level, fit=False)  # TODO: May want to cache this during original fit, as we do with OOF preds
+                        y_input = y_val
+
+                    # TODO: stack_name=REFIT_FULL_NAME_AUX?
+                    models_trained = self.generate_weighted_ensemble(X=X_train_stack_preds, y=y_input, level=level, stack_name=REFIT_FULL_NAME, kfolds=0, n_repeats=1, base_model_names=list(model.stack_column_prefix_to_model_map.values()), name_suffix=REFIT_FULL_PREFIX)
+                    # TODO: Do the below more elegantly, ideally as a parameter to the trainer train function to disable recording scores/pred time.
+                    for model_weighted_ensemble in models_trained:
+                        model_loaded = self.load_model(model_weighted_ensemble)
+                        model_loaded.val_score = None
+                        model_loaded.predict_time = None
+                        self.model_performance[model_weighted_ensemble] = None
+                        self.model_pred_times[model_weighted_ensemble] = None
+                        self.save_model(model_loaded)
+                else:
+                    models_trained = self.stack_new_level_core(X=X_full, y=y_full, models=[model_compressed], level=level, stack_name=REFIT_FULL_NAME, hyperparameter_tune=False, feature_prune=False, kfolds=0, n_repeats=1, stacker_type=stacker_type)
+                if len(models_trained) == 1:
+                    model_compressed_dict[model_name] = models_trained[0]
+                models_trained_full += models_trained
+
+        keys_to_del = []
+        for model in model_compressed_dict.keys():
+            if model_compressed_dict[model] not in models_trained_full:
+                keys_to_del.append(model)
+        for key in keys_to_del:
+            del model_compressed_dict[key]
+        self.model_compressed_dict.update(model_compressed_dict)
+        self.save()  # TODO: This could be more efficient by passing in arg to not save if called by refit_ensemble_full since it saves anyways later.
         return models_trained_full
 
+    # TODO: Rename _FULL -> _SINGLE / _SOLO?
```

Comment:
I think FULL actually sounds ok, since the biggest concern is that users realize the model has been trained on all of their data so they cannot view it's accuracies as unbiased.

Reply:
Agreed, removed TODO